isobar/pattern/sequence.py

```diff
@@ -197,14 +214,17 @@ class PLoop(Pattern):
         [1, 4, 9, 1, 4, 9, 1, 4, 9, 1, 4, 9, 1, 4, 9, 1]
         """
 
-    def __init__(self, pattern, count=sys.maxsize):
+    def __init__(self, pattern: Iterable, count: int = sys.maxsize):
```

Comment:
Just noticed that `pattern` is marked as an `Iterable` here and in a few other places later in the PR, but I think should always be a `Pattern`, right? (And is marked as a `Pattern` in most other places).

Reply:
This was mainly used in places that looked like a list could work as well as a pattern input, I wasn't sure if we wanted to keep things more broad for this and meant to run this by you on if we should keep everything as `Pattern` in this instance

Reply:
Sounds good, this change has been reflected in the newest commit. Should any non-list params that seem to reference Patterns (like `input` for `PMap` as seen below, or `PScaleLinLin`) also be type hinted as a pattern?

```py
class PMap(Pattern):
    """ PMap: Apply an arbitrary function to an input pattern.
              Will pass any additional arguments, which can also be patterns.
              Instances of None in the input stream will be passed through to
              the function as normal.

        >>> PMap(PSeries(), lambda value: value * value).nextn(16)
        [0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225]

        >>> PMap(PSeries(), pow, PSeries()).nextn(16)
        [1, 1, 4, 27, 256, 3125, 46656, 823543, 16777216, 387420489, 10000000000, ... ]
        """

    def __init__(self, input: Iterable, operator: Callable, *args, **kwargs):
```

Reply:
Yep, sounds perfect!

Reply:
Implemented this in the newest commit